cmd/ttn-lw-stack/commands/as_db.go

```diff
@@ -0,0 +1,176 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"crypto/tls"
+	"time"
+
+	pbtypes "github.com/gogo/protobuf/types"
+	"github.com/spf13/cobra"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io/web"
+	"go.thethings.network/lorawan-stack/v3/pkg/cluster"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/unique"
+	"google.golang.org/grpc"
+)
+
+func NewClusterComponentConnection(ctx context.Context, config Config, role ttnpb.ClusterRole) (*grpc.ClientConn, cluster.Cluster, error) {
+	clusterOpts := []cluster.Option{}
+	if config.Cluster.TLS {
+		tlsConf := config.TLS
+		tls := &tls.Config{
+			MinVersion:         tls.VersionTLS12,
+			InsecureSkipVerify: tlsConf.InsecureSkipVerify,
+		}
+		if err := tlsConf.Client.ApplyTo(tls); err != nil {
+			return nil, nil, err
+		}
+		clusterOpts = append(clusterOpts, cluster.WithTLSConfig(tls))
+	}
+	c, err := cluster.New(ctx, &config.Cluster, clusterOpts...)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.Join()
+	time.Sleep(20 * time.Millisecond)
```

Comment:
I think this is slightly optimistic, and instead we should look a couple of times the `GetPeerConn` (+delay).